figures/PNAS/npx/kCSD2D_reconstruction_from_npx.py

```diff
@@ -0,0 +1,133 @@
+    plt.figure(figsize=(12, 8))
+    # plt.suptitle('plane: '+str(k.estm_x[cut,0])+' $\mu$m '+' $\lambda$ : '+str(k.lambd)+
+                 # '  R: '+ str(k.R))
+    ax1 = plt.subplot(122)
+    set_axis(ax1, -0.05, 1.05, letter= 'D')
+    make_plot_spacetime(ax1, k.estm_x, k.estm_y, est_csd[cut,:,:], Fs,
+              title='Estimated CSD', cmap='bwr')
+    for lvl, name in zip([-500,-850,-2000], ['II/III', 'IV', 'V/VI']):
+        plt.axhline(lvl, ls='--', color='grey')
+        plt.text(340, lvl+20, name, fontsize=15)
+    plt.xlim(250, 400)
+    plt.xticks([250, 300, 350, 400], [-50, 0, 50, 100])
+    ax2 = plt.subplot(121)
+    set_axis(ax2, -0.05, 1.05, letter= 'C')
+    make_plot_spacetime(ax2, k.estm_x, k.estm_y, est_pots[cut,:,:],
+              title='Estimated LFP', cmap='PRGn')
+    plt.axvline(tp/Fs*1000, ls='--', color ='grey', lw=2)
+    plt.xlim(250, 400)
+    plt.xticks([250, 300, 350, 400], [-50, 0, 50, 100])
+    plt.tight_layout()
+    plt.savefig('figure_1D_pics', dpi=300)
+
+def plot_2D_pics(k, est_csd, est_pots, tp, Fs, cut, save=0):
+    plt.figure(figsize=(12, 8))
+    ax1 = plt.subplot(122)
+    set_axis(ax1, -0.05, 1.05, letter= 'B')
+    make_plot(ax1, k.estm_x, k.estm_y, est_csd[:,:,tp], 
+              title='Estimated CSD', cmap='bwr')
+    # for i in range(383): plt.text(ele_pos_for_csd[i,0], ele_pos_for_csd[i,1]+8, str(i+1))
+    plt.axvline(k.estm_x[cut][0], ls='--', color ='grey', lw=2)
+    ax2 = plt.subplot(121)
+    set_axis(ax2, -0.05, 1.05, letter= 'A')
+    make_plot(ax2, k.estm_x, k.estm_y, est_pots[:,:,tp],
+              title='Estimated LFP', cmap='PRGn')
+    # plt.suptitle(' $\lambda$ : '+str(k.lambd)+ '  R: '+ str(k.R))
+    plt.tight_layout()
+    plt.savefig('figure_2D_pics', dpi=300)
+
+def do_kcsd(ele_pos_for_csd, pots_for_csd, ele_limit):
+    ele_position = ele_pos_for_csd[:ele_limit[1]][0::1]
+    csd_pots = pots_for_csd[:ele_limit[1]][0::1]
+    k = KCSD2D(ele_position, csd_pots,
+               h=1, sigma=1, R_init=32, lambd=1e-9,
+               xmin= -42, xmax=42, gdx=4,
+               ymin=0, ymax=4000, gdy=4)
+    # k.L_curve(Rs=np.linspace(16, 48, 3), lambdas=np.logspace(-9, -3, 20))
+    return k, k.values('CSD'), k.values('POT'), ele_position
+#%%
+if __name__ == '__main__':
+    lowpass = 0.5
+    highpass = 300
+    Fs = 30000
+    resamp = 12
+    tp= 760
+    
+    forfilt=np.load('npx_data.npy')
+    
+    [b,a] = butter(3, [lowpass/(Fs/2.0), highpass/(Fs/2.0)] ,btype = 'bandpass')
+    filtData = filtfilt(b,a, forfilt)
+    pots_resamp = filtData[:,::resamp]
+    pots = pots_resamp[:, :]
+    Fs=int(Fs/resamp)
+    
+    pots_for_csd = np.delete(pots, 191, axis=0)
+    ele_pos_def = eles_to_coords(np.arange(384,0,-1))
+    ele_pos_for_csd = np.delete(ele_pos_def, 191, axis=0)
+    
+    k, est_csd, est_pots, ele_pos = do_kcsd(ele_pos_for_csd, pots_for_csd, ele_limit = (0,384))
+    
+    plot_1D_pics(k, est_csd, est_pots, tp, Fs, 15) 
```

Comment:
I should put the last argument of the function "15" together with other arguments at begining of the main block. My mistake but lets leave it maybe